doc/lenses.rst

```diff
@@ -3,16 +3,494 @@
 Lenses
 ======
 
-.. admonition:: Work in progress
-   :class: danger
+A tool borrowed from functional programming. While lenses are used
+extensively in Lager for zooming on cursors, they are also useful on
+their own for manipulating immutable data.
 
-   This section of the documentation has not been written yet.
+.. _making-a-lens:
 
-   Meanwhile, you can watch the **C++ Russia 2019 Talk**: `Squaring
-   the circle: value oriented design in an object oriented system
-   <https://www.youtube.com/watch?v=e2-FRFEx8CA>`_ (`slides`_).
+Making a Lens
```

Comment:
I've noticed you have been capitalising section headers Using Title Style. This is a bit inconsistent with the rest of the documentation.